packages/frontend/core/src/commands/affine-settings.tsx

```diff
@@ -36,7 +37,26 @@ export function registerAffineSettingsCommands({
       icon: <SettingsIcon />,
       run() {
         const quickSearchModalState = store.get(openQuickSearchModalAtom);
-        store.set(openQuickSearchModalAtom, !quickSearchModalState);
+
+        const editor = store.get(editorContainerAtom);
+        if (!editor) {
+          return store.set(openQuickSearchModalAtom, !quickSearchModalState);
+        }
+        // Due to a conflict with the shortcut for creating a link after selecting text in blocksuite,
+        // opening the quick search modal is disabled when text is selected.
+        const textSelection = editor.root?.std.selection.find('text');
+        if (textSelection && textSelection.from.length > 0) {
```

Comment:
You don't need to judge whether `TextSelection` exists here, because the format-bar might also appear in other selections.

Reply:
Ah, does the other formatBar also support creating links with cmdk?

Reply:
Are you satisfied with the modified result?